### PR TITLE
Ria 6647 ftpa in flight rework

### PR DESCRIPTION
**Pull request by hussain-faruk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-6647

### Change description ###
Added functionality to handle in flight cases where no ftpaApplicationDeadline is present for appellant/respondent ftpa applications.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaAppellantPreparerTest.java [hunk 1]
```diff
@@ -288,4 +318,54 @@ void should_not_allow_null_arguments() {
             .hasMessage("callback must not be null")
             .isExactlyInstanceOf(NullPointerException.class);
     }
+
+    @ParameterizedTest
+    @ValueSource(strings = {IN_FLIGHT_CASES_FTPA_OUT_OF_TIME_DATE, IN_FLIGHT_CASES_FTPA_IN_TIME_DATE_OOC})
+    void should_populate_ftpa_submission_out_of_time_to_correct_value_for_in_flight_cases_ooc(String input) {
```
Comment (neha-aggarw) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaAppellantPreparerTest.java:324]: Better naming convention for variable input would be appealDate. But change only if you are pushing another commit, else its fine.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaAppellantPreparerTest.java [hunk 2]
```diff
@@ -288,4 +318,54 @@ void should_not_allow_null_arguments() {
...
+            verify(asylumCase).write(FTPA_APPELLANT_SUBMISSION_OUT_OF_TIME, YesOrNo.YES);
+        } else {
+            verify(asylumCase).write(FTPA_APPELLANT_SUBMISSION_OUT_OF_TIME, YesOrNo.NO);
+        }
+
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {IN_FLIGHT_CASES_FTPA_OUT_OF_TIME_DATE, IN_FLIGHT_CASES_FTPA_IN_TIME_DATE_UK})
+    void should_populate_ftpa_submission_out_of_time_to_correct_value_for_in_flight_cases_uk(String input) {
```
Comment (neha-aggarw) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaAppellantPreparerTest.java:349]: Same as above.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaRespondentPreparerTest.java [hunk 3]
```diff
@@ -284,4 +316,53 @@ void should_not_allow_null_arguments() {
             .hasMessage("callback must not be null")
             .isExactlyInstanceOf(NullPointerException.class);
     }
+
+    @ParameterizedTest
+    @ValueSource(strings = {IN_FLIGHT_CASES_FTPA_OUT_OF_TIME_DATE, IN_FLIGHT_CASES_FTPA_IN_TIME_DATE_OOC})
+    void should_populate_ftpa_submission_out_of_time_to_correct_value_for_in_flight_cases_ooc(String input) {
```
Comment (neha-aggarw) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaRespondentPreparerTest.java:322]: Same as above.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaRespondentPreparerTest.java [hunk 4]
```diff
@@ -284,4 +316,53 @@ void should_not_allow_null_arguments() {
...
+            verify(asylumCase).write(FTPA_APPELLANT_SUBMISSION_OUT_OF_TIME, YesOrNo.YES);
+        } else {
+            verify(asylumCase).write(FTPA_APPELLANT_SUBMISSION_OUT_OF_TIME, YesOrNo.NO);
+        }
+
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {IN_FLIGHT_CASES_FTPA_OUT_OF_TIME_DATE, IN_FLIGHT_CASES_FTPA_IN_TIME_DATE_UK})
+    void should_populate_ftpa_submission_out_of_time_to_correct_value_for_in_flight_cases_uk(String input) {
```
Comment (neha-aggarw) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaRespondentPreparerTest.java:347]: Same as above.

**Review by neha-aggarw — APPROVED** [review]

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Pull request merged**